code/modules/antagonists/_common/antag_datum.dm

```diff
@@ -151,6 +155,7 @@ GLOBAL_LIST_EMPTY(antagonists)
 			info_button.Trigger()
 	apply_innate_effects()
 	give_antag_moodies()
+	RegisterSignal(owner.current, COMSIG_MINDSHIELDED_IMPLANTED, .proc/on_mindshield)
```

Comment:
If you're binding to the body not the mind then you need to handle mob changes. As is this would lead to leftover listeners.

Reply:
I'll set it to their mind instead, makes more sense for antag datums.